Jsonparser.cpp

```diff
@@ -0,0 +1,135 @@
+#include "Jsonparser.h"
+#include<sstream>
+#include<string>
+std::string Jsonparser:: determine_type(std::string str)
+{
+	
+	if (str.find('"') != std::string::npos)
+		return "string";
+	else
+	{
+		if (str.find('.') != std::string::npos)
+			return "double";
+		else
+			return "int";
+	}
+}
+std::map <std::string, std::string> Jsonparser::getmap(std::string str)
+{
+
+		std::map<std::string, std::string>m;
+		std::string line;
+
+		std::istringstream stream(str);
+
+		while (std::getline(stream, line))
+		{
+			std::string str1 = "";
+			std::string str2 = "";
+			std::string str0 = "";
+			std::string str00 = "";
+
+			bool whiteSpacesOnly = std::all_of(line.begin(), line.end(), isspace);
+
+			if (line.find('{') == std::string::npos && line.find('}') == std::string::npos && !whiteSpacesOnly)
+
+			{
+				/*determine key*/
+				str0 = line.substr(0, line.find(':') - 1);
+
+				str00 = str0.substr(str0.find('"') + 1, (str0.rfind('"') - str0.find('"')) - 1);
+	//			
+				std::string type = determine_type(line.substr(line.find(':')));
+
+				if (type == "string")
+				{
+					str1 = line.substr(line.find(':'));
+					str2 = str1.substr(str1.find('"') + 1, (str1.rfind('"') - str1.find('"')) - 1);
+					m[str00] = str2;
+				}
+				else
+				{
+					str1 = line.substr(line.find(':'));
+					str2 = str1.substr(str1.find(':') + 1, str1.rfind(',') - 1);
+					if (type == "int") {
+						m[str00] = str2;
+					}
+					else {
+						if (type == "double")
+						{
+							m[str00] = str2;
+						}
+					}
+				}
+
+			}
+		}
+			return m;
+}
+std::map <std::string, std::string> Jsonparser::parseJson(std::string fname)
+{
```

Comment:
Most ide is bemásoltad majdnem pontosan ugyanazt, amit fentebb leírtál? Ne csinálj ilyet! Ne legyen redundáns a kód!